analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -1394,12 +1398,21 @@ ProjectSettings getSettings() {
     return downloadedSettings;
   }
 
-  void performInitializeIntegrations(ProjectSettings projectSettings) {
+  void performInitializeIntegrations(ProjectSettings projectSettings) throws IOException {
+    if (isNullOrEmpty(projectSettings)) {
+      throw new IOException("ProjectSettings is empty!");
+    }
     ValueMap integrationSettings = projectSettings.integrations();
+    if (isNullOrEmpty(integrationSettings)) {
```

Comment:
I think it's ok for this to null. If it's null, we can treat this is the same as empty map (which would represent the case when no client side integrations are enabled).